src/tmpfiles/tmpfiles.c

```diff
@@ -212,6 +214,29 @@ static const Specifier specifier_table[] = {
         {}
 };
 
+static void setup_specifier_table(const char *prefix) {
+        int r, i;
+        int prohibit_runtime = 0;
+
+        r = getenv_bool("SYSTEMD_SPECIFIER_PROHIBIT_RUNTIME");
+        if (r > 0)
+                prohibit_runtime = r;
+        if (!prohibit_runtime || !prefix)
+                return;
+
+        for (i = 0; i < (int) ELEMENTSOF(specifier_table) - 1; ++i) {
+                const char spec = specifier_table[i].specifier;
+                if (memchr("mowBW", spec, sizeof("mowBW") - 1))
+                        /* specifier takes an argument for root prefix */
+                        specifier_table[i].data = prefix;
+                else if (spec == 'T' || spec == 'V') {
+                        specifier_table[i].lookup = specifier_string;
+                        specifier_table[i].data = spec == 'T' ? "/tmp" : "/var/tmp";
+                } else if (specifier_table[i].lookup != specifier_directory)
+                        specifier_table[i].lookup = disabled_specifier;
```

Comment:
instead of patching around in the specifier table, make them constant please, but have two around, one for the no-runtime case and one for the normal case. Yes, they will be mostly idnetical, but that's fine.

Reply:
I can't make them constant, unless I use some static array for `prefix`. Otherwise the pointer to prefix will only be known at runtime

Reply:
specifier_printf() has a third param, `userdata`, we only pass `NULL` there so far. Pass the `prefix` there instead.